app/src/main/java/com/ciandt/dragonfly/example/components/chips/ChipsView.kt

```diff
@@ -0,0 +1,140 @@
+    }
+
+    fun initializeAttributes(attrs: AttributeSet) {
+
+        val typedArray = context.obtainStyledAttributes(attrs, R.styleable.ChipsView, 0, 0)
+        try {
+
+            val chipLayout = typedArray.getResourceId(R.styleable.ChipsView_chipLayout, -1)
+            if (chipLayout != -1) {
+                setChipLayout(chipLayout)
+            }
+
+            val chipSpaceBefore = typedArray.getDimensionPixelSize(R.styleable.ChipsView_chipSpaceBefore, DEFAULT_SPACE_BEFORE)
+            val chipSpaceBetween = typedArray.getDimensionPixelSize(R.styleable.ChipsView_chipSpaceBetween, DEFAULT_SPACE_BETWEEN)
+            val chipSpaceAfter = typedArray.getDimensionPixelSize(R.styleable.ChipsView_chipSpaceAfter, DEFAULT_SPACE_AFTER)
+            setChipSpace(chipSpaceBefore, chipSpaceBetween, chipSpaceAfter)
+
+            val fadingEdges = typedArray.getBoolean(R.styleable.ChipsView_fadingEdges, false)
+            setFadingEdges(fadingEdges)
+
+            val fadingEdgesDrawable = typedArray.getDrawable(R.styleable.ChipsView_fadingEdgesDrawable)
+            setFadingEdgesDrawable(fadingEdgesDrawable)
+
+            val fadingEdgesWidth = typedArray.getDimensionPixelSize(R.styleable.ChipsView_fadingEdgesWidth, DEFAULT_EDGES_WIDTH)
+            setFadingEdgesWidth(fadingEdgesWidth)
+
+            val selectable = typedArray.getBoolean(R.styleable.ChipsView_selectable, false)
+            setSelectable(selectable)
+
+        } finally {
+            typedArray.recycle()
+        }
+    }
+
+    fun setChipLayout(layout: Int) {
+        adapter.setChipLayout(layout)
+    }
+
+    fun setChipSpace(spaceBefore: Int, spaceBetween: Int, spaceAfter: Int) {
+        recyclerView.addItemDecoration(SpaceItemDecoration(spaceBefore, spaceBetween, spaceAfter))
+    }
+
+    fun setFadingEdges(fadingEdges: Boolean) {
+        fadingStart.visibility = if (fadingEdges) View.VISIBLE else View.INVISIBLE
+        fadingEnd.visibility = if (fadingEdges) View.VISIBLE else View.INVISIBLE
+    }
+
+    fun setFadingEdgesDrawable(fadingEdgesDrawable: Drawable?) {
+        fadingEdgesDrawable?.let {
+            fadingStart.setImageDrawable(it)
+            fadingEnd.setImageDrawable(it)
+        }
+    }
+
+    fun setFadingEdgesWidth(fadingEdgesWidth: Int) {
+        fadingStart.layoutParams.width = fadingEdgesWidth
+        fadingEnd.layoutParams.width = fadingEdgesWidth
+    }
+
+    fun setSelectable(selectable: Boolean) {
+        adapter.setSelectable(selectable)
+    }
+
+    fun setChips(chips: List<Chip>) {
+        this.chips.clear()
+        this.chips.addAll(chips)
+        adapter.notifyItemInserted(chips.size - 1)
+    }
+
+    fun setSelectCallback(callback: ((Chip) -> Unit)?) {
+        selectCallback = callback
+    }
+
+    fun setDeselectCallback(callback: ((Chip) -> Unit)?) {
+        deselectCallback = callback
+    }
+
+    private fun dpToPx(dp: Int): Int = (dp * Resources.getSystem().displayMetrics.density).toInt()
```

Comment:
What about this as an extension function to View or another framework class? Feels like a potentially reusable logic.